@here/harp-mapview/lib/poi/PoiManager.ts

```diff
@@ -491,7 +491,13 @@ export class PoiManager {
     ): TextElement {
         const priority = technique.priority !== undefined ? technique.priority : 0;
         const positions = Array.isArray(x) ? (x as THREE.Vector3[]) : new THREE.Vector3(x, y, z);
+
+        // The current zoomlevel of mapview. Since this method is called for all tiles in the
+        // VisibleTileSet we can be sure that the current zoomlevel matches the zoomlevel where
+        // the tile should be shown.
         const displayZoomLevel = this.mapView.zoomLevel;
+        const floorZoomLevel = Math.floor(displayZoomLevel);
+        assert(floorZoomLevel === tile.tileKey.level - tile.dataSource.storageLevelOffset);
```

Comment:
why do you have this assertion here? This would point to some issue in the VisibleTileSet?

Reply:
The assert is here to make it more clear and that the following code is relying on it.

Reply:
When you mean the following code relies on it, you mean that the getPropertyValue method relies on the correct zoom level to give the correct value?

It just seems strange to have this check here, and not somewhere (for example in the VisibleTileSet), or in the render method on the MapView. Because the check only needs to be done one for each tile, and not for each POI in each tile, perhaps straight after the line:

```
renderedTiles.forEach(tile => {
```

Second question, this method is called when creating the poi (as far as I can tell), why does the MapView zoom level influence the creation? Just seems like a strange dependency to have.

Reply:
You are right since we don't have a proper production build that removes asserts this could have some performance impact.
Regarding using the zoomlevel here: You are right the logic is broken. All properties should be evaluate every frame but with the current state of the label rendering everything would fall apart. Especially changing properties that drastically change the layout (like vAlignment and hAlignment) should not change between zoomlevels b/c it will flicker even more. Also the performance of our interpolants is really bad right now. @robertoraggi is refactoring them and once they are performant we can also refactor this code.

Reply:
Ok, at least it is known that this is broken and that it is in the pipeline to be fixed.